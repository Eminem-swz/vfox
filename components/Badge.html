<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Badge 徽标数 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.133cf44e.css">
    <link rel="modulepreload" href="/vfox/assets/Home.999026eb.js">
    <link rel="modulepreload" href="/vfox/assets/app.3c8a5e7e.js">
    <link rel="modulepreload" href="/vfox/assets/components_Badge.md.da763bd6.lean.js">
    
    <meta name="twitter:title" content="Badge 徽标数 | Vfox for Vue3">
  <meta property="og:title" content="Badge 徽标数 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-3eed0030><div class="sidebar-button" data-v-3eed0030><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-3eed0030 data-v-d91ad9e0><!----> Vfox for Vue3</a><div class="flex-grow" data-v-3eed0030></div><div class="nav" data-v-3eed0030><nav class="nav-links" data-v-3eed0030 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/playground/" target="_blank" data-v-a00c0156>Playground <!----></a></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-1314d532><nav class="nav-links nav" data-v-1314d532 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/playground/" target="_blank" data-v-a00c0156>Playground <!----></a></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-1314d532><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Badge">Badge</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#slots">Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#默认（-default）">默认（#default）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-33252f1f><div class="container" data-v-33252f1f><!--[--><!--[--><!----><!--]--><!--[--><div class="demo"><iframe style="" src="/vfox/demo/#/Badge" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-33252f1f><div><h1 id="badge-徽标数" tabindex="-1">Badge 徽标数 <a class="header-anchor" href="#badge-徽标数" aria-hidden="true">#</a></h1><p>注：</p><ul><li>这是个包裹组件，需要 <code>slot</code> 中提供被挂在的内容。</li></ul><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-JavaScript"><pre><code><span class="token keyword">import</span> <span class="token punctuation">{</span> Badge <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">&#39;vfox&#39;</span>
</code></pre></div><p>具体的引入方式可以参考<a href="./../guide/import.html">引入组件</a>。</p><h2 id="props" tabindex="-1">Props <a class="header-anchor" href="#props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>content</td><td>number | string</td><td>0</td><td>否</td><td>消息条数或者文本</td></tr><tr><td>max-count</td><td>number</td><td>99</td><td>否</td><td>最大完全显示消息条数</td></tr><tr><td>dot</td><td>boolean</td><td>false</td><td>否</td><td>是否显示为小红点</td></tr><tr><td>show-zero</td><td>boolean</td><td>false</td><td>否</td><td>消息数为 0 时是否显示</td></tr><tr><td>offset</td><td>number[]</td><td>[0, 0]</td><td>否</td><td>偏移量，格式为 [x, y]</td></tr><tr><td>animated</td><td>boolean</td><td>false</td><td>否</td><td>是否开启动画</td></tr><tr><td>color</td><td>string</td><td></td><td>否</td><td>自定义色彩，支持 hex rgb hsl 等写法，详细效果<a href="./../design/color.html#自定义色彩">查看</a></td></tr></tbody></table><h2 id="slots" tabindex="-1">Slots <a class="header-anchor" href="#slots" aria-hidden="true">#</a></h2><h3 id="默认（-default）" tabindex="-1">默认（#default） <a class="header-anchor" href="#默认（-default）" aria-hidden="true">#</a></h3><div class="language-Vue"><pre><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-badge</span> <span class="token attr-name">:count</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>1<span class="token punctuation">&quot;</span></span> <span class="token attr-name">show-zero</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-button</span><span class="token punctuation">&gt;</span></span>badge<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-button</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-badge</span><span class="token punctuation">&gt;</span></span>
</code></pre></div></div></div><footer class="page-footer" data-v-33252f1f data-v-c772b490><div class="edit" data-v-c772b490><div class="edit-link" data-v-c772b490 data-v-11add020><!----></div></div><div class="updated" data-v-c772b490><!----></div></footer><div class="next-and-prev-link" data-v-33252f1f data-v-6889d424><div class="container" data-v-6889d424><div class="prev" data-v-6889d424><a class="link" href="/vfox/components/Divider" data-v-6889d424><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-6889d424><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-6889d424>Divider</span></a></div><div class="next" data-v-6889d424><a class="link" href="/vfox/components/Progress" data-v-6889d424><span class="text" data-v-6889d424>Progress</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-6889d424><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"components_actionsheet.md\":\"a219a50b\",\"components_activityindicator.md\":\"e9478325\",\"components_avatar.md\":\"df4d6f34\",\"components_backtop.md\":\"44951bf2\",\"components_badge.md\":\"da763bd6\",\"components_button.md\":\"e4a97915\",\"components_calendar.md\":\"419582e7\",\"components_cascader.md\":\"ba25b1f9\",\"components_cell.md\":\"9bceb72a\",\"components_checkbox.md\":\"e66443b1\",\"components_circleprogress.md\":\"a577b6da\",\"components_collapse.md\":\"6f2bd5df\",\"components_copy.md\":\"ef2c8fc0\",\"components_countdown.md\":\"cc67e5dc\",\"components_countup.md\":\"f58eb9e7\",\"components_datepicker.md\":\"bbdf2abe\",\"components_dialog.md\":\"b9d53062\",\"components_divider.md\":\"a21d276f\",\"components_drawer.md\":\"008d8ccd\",\"components_dropdown.md\":\"4497ff66\",\"components_empty.md\":\"629924a9\",\"components_fixed.md\":\"e02275dd\",\"components_flatlist.md\":\"7f21ab51\",\"components_form.md\":\"54b694a5\",\"components_group.md\":\"54fcb08d\",\"components_icon.md\":\"13291897\",\"components_image.md\":\"39038cb7\",\"components_imagepreview.md\":\"57aaf845\",\"components_imageuploader.md\":\"98f68c3e\",\"components_indexview.md\":\"448a59a2\",\"components_input.md\":\"e42aa609\",\"components_layout.md\":\"60b3c4c5\",\"components_loadmore.md\":\"58af78e7\",\"components_modal.md\":\"832786bc\",\"components_navbar.md\":\"afd87045\",\"components_noticebar.md\":\"a865f2c2\",\"components_notify.md\":\"e4202c1f\",\"components_numberkeyboard.md\":\"051760d5\",\"components_order.md\":\"ca2d235e\",\"components_pagination.md\":\"67c6dbf5\",\"components_picker.md\":\"b0e6360a\",\"components_popdialog.md\":\"4a7d7ffe\",\"components_popmenu.md\":\"40593c23\",\"components_popover.md\":\"41d3bfcd\",\"components_price.md\":\"40b4d597\",\"components_progress.md\":\"9659f92d\",\"components_radio.md\":\"c358fabe\",\"components_range.md\":\"2298b612\",\"components_rate.md\":\"21125ddc\",\"components_result.md\":\"8b1fa6d6\",\"components_scrolltab.md\":\"af65ce57\",\"components_scrollview.md\":\"7a1e4622\",\"components_searchbar.md\":\"0bc6211c\",\"components_sidetab.md\":\"3214eacc\",\"components_skeleton.md\":\"4836db7d\",\"components_slider.md\":\"e44e6ad4\",\"components_stepper.md\":\"d64b5d92\",\"components_steps.md\":\"6968cea7\",\"components_sticky.md\":\"c94fbd00\",\"components_stopwatch.md\":\"6e16bbf8\",\"components_swipecell.md\":\"8c356658\",\"components_swiper.md\":\"e391ebec\",\"components_switch.md\":\"818cc24b\",\"components_tab.md\":\"2d98d99a\",\"components_tabbar.md\":\"63edfd7d\",\"components_tabview.md\":\"2e96c595\",\"components_tag.md\":\"c349be11\",\"components_timeago.md\":\"301526b4\",\"components_timeline.md\":\"ebedc212\",\"components_toast.md\":\"8ec2b419\",\"components_index.md\":\"9f329393\",\"design_color.md\":\"7c64cbc3\",\"design_index.md\":\"3e2ce150\",\"guide_dark.md\":\"b8f9ecbe\",\"guide_import.md\":\"39c5e512\",\"guide_locale.md\":\"6c36d029\",\"guide_theme.md\":\"78169609\",\"index.md\":\"326a44f8\"}")</script>
    <script type="module" async src="/vfox/assets/app.3c8a5e7e.js"></script>
    
  </body>
</html>