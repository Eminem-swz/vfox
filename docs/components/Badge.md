# Badge 徽标数

注：

- 这是个包裹组件，需要 `slot` 中提供被挂在的内容。

## Import

```
import { Badge } from 'vfox'
```

具体的引入方式可以参考[引入组件](../index.md#引入组件)。

## Props

| 属性      | 类型          | 默认值 | 必填 | 说明                                                                               |
| --------- | ------------- | ------ | ---- | ---------------------------------------------------------------------------------- |
| content   | number/string | 0      | 否   | 消息条数或者文本                                                                   |
| max-count | number        | 99     | 否   | 最大完全显示消息条数                                                               |
| dot       | boolean       | false  | 否   | 是否显示为小红点                                                                   |
| show-zero | boolean       | false  | 否   | 消息数为 0 时是否显示                                                              |
| offset    | number[]      | [0, 0] | 否   | 偏移量，格式为 [x, y]                                                              |
| animated  | boolean       | false  | 否   | 是否开启动画                                                                       |
| color     | string        |        | 否   | 自定义色彩，支持 hex rgb hsl 等写法，详细效果[查看](../design/color.md#自定义色彩) |

## Slots

### 默认（#default）

```
<fx-badge :count="1" show-zero>
  <fx-button>badge</fx-button>
</fx-badge>
```
