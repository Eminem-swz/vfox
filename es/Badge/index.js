import { defineComponent, ref, computed, watch, onBeforeUnmount, openBlock, createBlock, renderSlot, toDisplayString, createCommentVNode } from 'vue';

/**
 * 将字段名转为驼峰式格式
 * @param {string} name 字段名
 */
/**
 * 是否函数
 * @param object 值
 */
function isFunction(object) {
    return typeof object === 'function';
}
/**
 * 是否字符串
 * @param object 值
 */
const isString = (object) => {
    return typeof object === 'string';
};
/**
 * 是否数值，这里会对排除无穷大/无穷小的情况
 * @param object 值
 */
function isNumber(object) {
    return typeof object === 'number' && isFinite(object);
}
/**
 * 是否数值/可转为数值，如'1.1'，这里会对排除无穷大/无穷小的情况
 * @param object 值
 */
const isNumeric = (object) => {
    return (isNumber(object) ||
        (isString(object) &&
            !isNaN(parseFloat(object)) &&
            isFinite(parseFloat(object))));
};
/**
 * 获取限定范围的数值
 * @param number 数值
 * @param min 最小值
 * @param max 最大值
 */
function rangeNumber(number, min, max) {
    return Math.min(max, Math.max(min, number));
}
/**
 * 获取限定范围内的整数
 * @param number 数值
 * @param min 最小值
 * @param max 最大值
 */
function rangeInteger(number, min, max) {
    if (isString(number)) {
        if (isNumeric(number)) {
            number = parseFloat(number);
        }
        else {
            number = min;
        }
    }
    return rangeNumber(Math.round(number), Math.ceil(min), Math.floor(max));
}

const Easing = {
    linear(p) {
        return p;
    },
    swing(p) {
        return 0.5 - Math.cos(p * Math.PI) / 2;
    }
};
class AnimationFrameTask {
    constructor(ref) {
        this.stop = function () {
            if (ref.idle) {
                cancelAnimationFrame(ref.idle);
                ref.idle = null;
            }
        };
    }
}
/**
 * 变化解帧
 * @param options 设置项
 */
function frameTo(options) {
    const { from, to, duration, progress, complete } = options;
    const start = Date.now();
    const end = start + duration;
    const ref = { idle: null };
    let frameIndex = 0;
    function step() {
        ref.idle = requestAnimationFrame(function () {
            const t = Date.now();
            let current;
            if (t >= end) {
                current = to;
                progress({
                    current,
                    frameIndex: frameIndex++
                });
                isFunction(complete) && complete({ current });
            }
            else {
                const p = Easing['swing']((t - start) / duration);
                current = from + (to - from) * p;
                if (isFunction(progress)) {
                    progress({
                        current,
                        frameIndex: frameIndex++
                    });
                }
                step();
            }
        });
    }
    if (from !== to) {
        step();
    }
    else {
        isFunction(complete) && complete({ current: to });
    }
    return new AnimationFrameTask(ref);
}

var script = defineComponent({
    name: 'fx-badge',
    props: {
        // 消息条数
        content: {
            type: [String, Number],
            default: 0
        },
        // 最大完全显示消息条数
        maxCount: {
            type: Number,
            default: 99
        },
        // 是否显示为小红点
        dot: {
            type: Boolean,
            default: false
        },
        // 消息数为0时是否显示
        showZero: {
            type: Boolean,
            default: false
        },
        // 是否开启动画
        animated: {
            type: Boolean,
            default: false
        },
        // 偏移量，格式为 [x, y]
        offset: {
            type: Array,
            default: () => {
                return [0, 0];
            }
        }
    },
    setup(props) {
        const content2 = ref(isString(props.content)
            ? props.content
            : isNumber(props.content)
                ? rangeInteger(props.content, 0, props.maxCount)
                : 0);
        let frameTask;
        const showCount = computed(() => {
            if (isString(content2.value)) {
                return content2.value;
            }
            if (props.content > props.maxCount && content2.value === props.maxCount) {
                return content2.value + '+';
            }
            return content2.value.toString();
        });
        const styles = computed(() => {
            return {
                transform: `translate3d(50%, -50%, 0px) scale(${(isString(props.content) && props.content) ||
                    props.showZero ||
                    props.content > 0
                    ? 1
                    : 0})`,
                right: `${-props.offset[0]}px`,
                top: `${props.offset[1]}px`
            };
        });
        watch(() => props.content, val => {
            frameTask && frameTask.stop();
            if (isString(val)) {
                content2.value = val;
                return;
            }
            if (!isNumber(val)) {
                return;
            }
            const currentIsShow = props.showZero || content2.value > 0;
            const isReadyToHide = !props.showZero && val === 0;
            if (!currentIsShow || isReadyToHide) {
                content2.value = val;
            }
            else {
                const to = rangeInteger(val, 0, props.maxCount);
                frameTask = frameTo({
                    from: content2.value,
                    to,
                    duration: Math.min(Math.abs(to - content2.value) * 50, 1000),
                    progress: ({ current, frameIndex }) => {
                        if (frameIndex % 3 === 0) {
                            content2.value = Math.round(current);
                        }
                    },
                    complete: ({ current }) => {
                        content2.value = current;
                    }
                });
            }
        });
        onBeforeUnmount(() => frameTask && frameTask.stop());
        return {
            showCount,
            styles
        };
    }
});

function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createBlock("div", {
    class: ["fx-badge", { animated: !!_ctx.animated }]
  }, [
    renderSlot(_ctx.$slots, "default"),
    (_ctx.dot && _ctx.content != null)
      ? (openBlock(), createBlock("i", {
          key: 0,
          class: "fx-badge_dot",
          style: _ctx.styles
        }, null, 4 /* STYLE */))
      : (_ctx.content != null)
        ? (openBlock(), createBlock("span", {
            key: 1,
            class: "fx-badge_num",
            style: _ctx.styles
          }, toDisplayString(_ctx.showCount), 5 /* TEXT, STYLE */))
        : createCommentVNode("v-if", true)
  ], 2 /* CLASS */))
}

script.render = render;
script.__file = "src/Badge/Badge.vue";

const _Badge = Object.assign(script, {
    install: function (app) {
        app.component(script.name, script);
    }
});

export default _Badge;
