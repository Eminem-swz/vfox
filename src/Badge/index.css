.fx-badge {
  display: inline-block;
  position: relative;
  vertical-align: top;
  --badge-background-color: #ff4d4f;
  --badge-color: #fff;
}

.fx-badge_num {
  position: absolute;
  height: 15px;
  line-height: 15px;
  min-width: 15px;
  display: inline-flex;
  align-items: center;
  padding: 0 4px;
  border-radius: 7.5px;
  box-sizing: border-box;
  font-size: 11px;
  font-weight: 400;
  color: var(--badge-color);
  background-color: var(--badge-background-color);
  transform: scale(0);
}

.fx-badge_dot {
  position: absolute;
  right: -5px;
  top: -5px;
  height: 10px;
  width: 10px;
  border-radius: 5px;
  box-sizing: border-box;
  color: var(--badge-color);
  background-color: var(--badge-background-color);
  border: 1px solid var(--border-color);
  transform: scale(0);
}

.fx-badge.animated .fx-badge_num {
  transition: transform 0.2s;
}

.fx-badge.animated .fx-badge_dot {
  transition: transform 0.1s;
}
