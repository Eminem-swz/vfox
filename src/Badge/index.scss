@import '../style/var.scss';

.fx-badge {
  display: inline-block;
  position: relative;
  vertical-align: top;
  --badge-background-color: #{$danger-color};
  --badge-color: #fff;

  &_badge {
    position: absolute;
    height: 17px;
    min-width: 17px;
    line-height: 17px;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    padding: 0 4px;
    border-radius: 8.5px;
    box-sizing: border-box;
    font-size: 11px;
    font-weight: 400;
    color: var(--badge-color);
    background-color: var(--badge-background-color);
    transform: scale(0);
    border: 1px solid #fff;
    box-sizing: border-box;

    .fx-icon {
      --size: 11px;
      --color: var(--badge-color);
      margin: 0 -4px;
    }

    &.dot {
      right: -6px;
      top: -6px;
      height: 12px;
      width: 12px;
      min-width: 0;
      padding: 0;
      overflow: hidden;
      text-indent: -9999px;

      * {
        display: none;
      }
    }
  }

  &.animated {
    .fx-badge {
      &_badge {
        transition: transform 0.2s;
      }
    }
  }
}
