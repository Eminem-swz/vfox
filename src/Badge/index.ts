import { App } from 'vue'
import { SFCWithInstall } from '@/helpers/types'
import Badge from './Badge.vue'

const _Badge: SFCWithInstall<typeof Badge> = Object.assign(Badge, {
  install: function(app: App) {
    app.component(Badge.name, Badge)
  }
})

export default _Badge
