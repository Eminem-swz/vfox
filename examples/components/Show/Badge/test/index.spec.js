/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Badge from '@/components/Badge'

describe('Badge', () => {
  test('should render prop content=1 correctly', () => {
    const wrapper = mount(Badge, {
      propsData: {
        content: 1
      },
      slots: {
        default: 'content'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
