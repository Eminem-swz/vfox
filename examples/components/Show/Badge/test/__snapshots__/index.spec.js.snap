// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render default slot correctly 1`] = `
<div class="fx-badge">
  <div class="fx-badge_badge" style="transform: translate3d(50%, -50%, 0px) scale(0); right: 0px; top: 0px;">0</div>
</div>
`;

exports[`Badge should render prop content=1 correctly 1`] = `
<div class="fx-badge">
  <div class="fx-badge_badge" style="transform: translate3d(50%, -50%, 0px) scale(1); right: 0px; top: 0px;">1</div>
</div>
`;

exports[`Badge should render prop content=1000 correctly 1`] = `
<div class="fx-badge">
  <div class="fx-badge_badge" style="transform: translate3d(50%, -50%, 0px) scale(1); right: 0px; top: 0px;">99+</div>
</div>
`;

exports[`Badge should render prop dot=true correctly 1`] = `
<div class="fx-badge">
  <div class="fx-badge_badge dot" style="transform: translate3d(50%, -50%, 0px) scale(1); right: 0px; top: 0px;">1</div>
</div>
`;
